### ISSUE TITLE
Collaboration Cycle for [Conditions Team, 21-526ez Disability Benefits Application, Classifier integration with ML models]

### ISSUE THREAD

**Issue by yurenanava** [department-of-veterans-affairs/va.gov-team#107299]
Please view the [Milestone](https://github.com/department-of-veterans-affairs/va.gov-team/milestone/1535) for all feedback and findings tickets associated with this Collaboration Cycle initiative. 
 
### VFS team name

Conditions Team

### Product name

21-526ez Disability Benefits Application

### Feature name

Contention Classifier Migration

### GitHub label for team

contention-classification

### GitHub label for product

21-526EZ

### Public DSVA Slack channel for VFS team

#benefits-contention-classification

## Kickoff questions
<details><summary>Toggle kickoff questions</summary>

### When did/will you start working on this product/feature?

February 6, 2023

### Will your work result in visible changes to the user experience?

No

### Where will your product live?

VA.gov

### Are you iterating on an existing product or is this a new product?

Iterating on an existing product

### Will your work involve changes to...

Other

### Are you doing research with VA.gov users?

No

### Does your product/feature have Google Analytics tracking and a KPI dashboard in Domo?

No

### Do you need to capture any additional analytics or metrics?

Yes

### Will a VA editor (Drupal) notice this change?

No

### Does your work require non-trivial code changes or involve PII/PHI?

Unsure

### Product outline

https://github.com/department-of-veterans-affairs/vagov-claim-classification/tree/main/collab-cycle/202504%20Classification%20using%20ML%20models#iteration-4-machine-learning-contention-text-classification

### Verify all relevant materials provided to Governance Team

- [x] I have provided all relevant and up-to-date links, screenshots, images, designs, etc. of the as-is version of the product

### Add the GitHub labels for your team, product, and feature to this ticket.

- [x] I acknowledge that I must add the GitHub labels for my team, product, and feature to this ticket.

Kickoff Slack Thread with VFS team: [Kickoff thread](https://dsva.slack.com/archives/C04AZ8T7XN1/p1744388612378859)
</details>

### Architecture Intent
<details>
  <summary>Toggle Architecture Intent instructions</summary>

##### VFS team actions
- [x] Review [Architecture Intent Guidance](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/architecture-intent) to understand what this touchpoint involves.
- [x] Schedule your Architecture Intent (with at least 2 business days lead time from now):
  - Open the [Calendly Architecture intent calendar](https://calendly.com/collaboration-cycle/architecture-intent)
  - Select a date and time and click "Confirm"
  - Add your name and email
  - Click "Add Guests" and enter the email addresses for VFS team attendees
  - Click "Schedule Event"
- [x] Link all artifacts in the `Architecture Intent artifacts for review` section below at least two business days before the scheduled Architecture Intent. Please don't add artifacts in the comments section.
  
**Architecture Intent artifacts for review**

[See guidance on Architecture Intent artifacts.](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/architecture-intent#ArchitectureIntent-Whatartifactswillbeneeded?) Feedback is based on the artifacts provided here as well as information provided during the meeting. Please provide links to artifacts **at least two business days** before the scheduled meeting.

Required:
- [x] Fully completed [Architecture Intent Meeting Template](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/engineering/collaboration-cycle/architecture-intent/checklist/Conditions%20Team%2C%20526ez%20Disability%20Benefits%20Application%2C%20Classifier%20integration%20with%20ML%20models%20-%2020250414.md
- [x] [User data flow diagram](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1677003925622/f5284353833891a9f341413073b3b16695026cba?wid=0-1738962400807), if your application accepts data from a user.

##### Platform Team actions
- [x] Meeting date/time: ~~04:30pm - Monday, June 23, 2025~~

##### Platform Team actions

- [ ] Update this ticket with the Zoom recording
  - Recording link
  - Passcode:
- Engineering feedback
  - [ ] Feedback added to milestone
  - [ ] No feedback
- Mobile Engineering feedback
  - [ ] Feedback added to milestone
  - [ ] No feedback
- Security feedback
  - [ ] Feedback added to milestone
  - [ ] No feedback  

##### VFS team actions

- [ ] Review feedback tickets. Comment on the ticket if there are any questions or concerns.
- [ ] [Complete Collaboration Cycle feedback survey](https://ows.io/qs/gb1dfnkl)

</details>

### Staging Review
<details>
  <summary>Toggle Staging Review</summary>
  
##### VFS actions
- Navigate to reference link: [Staging Review Guidance](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/staging-review)  
- [x] Schedule your Staging Review when ready:
   - Open the [Calendly staging review calendar](https://calendly.com/collaboration-cycle/staging-review)
   - Select a date and time and click “Confirm”
   - Add your name and email
   - Click "Add Guests" and enter the VFS meeting attendees email addresses
      - Invite all relevant VFS team members, including accessibility support, product owners, and other VA stakeholders
   - Click "Schedule Event"
- [ ] If this product contains any [experimental design](https://design.va.gov/about/contributing-to-the-design-system/experimental-components-and-patterns), add the `experimental-design` label and schedule a meeting with DSC to present the research findings.
- [ ] Link all artifacts **ONLY** in the Staging Review artifacts section below at least four days before the scheduled Staging Review.  **Do NOT add artifacts to Comments section**
- [ ] I confirm the environment is available and test users have been provided.

**Staging Review artifacts**

Links to [Staging Review artifacts](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Staging-review.1810137181.html#Stagingreview-Artifacts) must be added to this ticket 4 business days ahead of the scheduled meeting. Please do not make changes to the product or artifacts during the 4-day review period. 

**Required artifacts**
- [ ] **Direct link or instructions on how to access the product**
  - The product should be available on an [approved staging environment](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/staging-environment-guidance-for-vfs-teams). 
  - List screens, sections of screens, and/or user flows impacted by this work.
- [ ] **Generated PDF form** (when applicable)
  - A populated version of the form that will be submitted through the online tool 
- [ ] **Drupal or Staging URL for updated primary entry point:** the main way Veterans will access the tool through site navigation (not search)
  - If the primary entry point is not a page on VA.gov, include information about how to view it. Reach out to `@platform-governance-team-members` on Slack with any questions.
- [ ] **Test users and scenarios** (when applicable)
  - Store [test user information](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-accessible-example.md), passwords, and tasks in a .md file in the va.gov-team-sensitive repository. 
  - Make sure all user scenarios can be tested, i.e.: in-progress form, submitted form, new form.
- [ ] List of known issues and bugs
  - Include a list of known issues and/or bugs that might be flagged at Staging Review.
  - When possible, please include links to relevant tickets.
- [ ] **Link to Sitewide CAIA intake ticket**, if applicable.
- [ ] **Completed accessibility testing artifact:** see [instructions and link to accessibility testing template](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review).
- [ ] **QA Artifacts:** artifacts that correspond to each of the [QA Standards](https://depo-platform-documentation.scrollhelp.site/developer-docs/quality-assurance-standards). 
   - QA3, QA5, QA6: https://github.com/department-of-veterans-affairs/vagov-claim-classification/blob/main/collab-cycle/202504%20Classification%20using%20ML%20models/README.md#traceability-reports
  - [ ] Regression test plan
  - [ ] Test plan
  - [ ] Coverage for References
  - [ ] Summary (Defects) reports
  - [ ] E2E tests
  - [ ] Code coverage
  - [ ] Endpoint monitoring playbook
  - [ ] Logging silent failures
- [x] Completed **[Engineering and Security Checklist](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/engineering/collaboration-cycle/architecture-intent/checklist/eng-sec-checklist.md)**
     - [Conditions Team, 21-526ez Disability Benefits Application, Classifier integration with ML models - 09292025.md](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/engineering/collaboration-cycle/architecture-intent/checklist/Conditions%20Team%2C%2021-526ez%20Disability%20Benefits%20Application%2C%20Classifier%20integration%20with%20ML%20models%20-%2009292025.md)

**Not required, but nice to have artifacts**
- [ ] **Content source of truth:** link to CAIA Content feedback, such as a content source of truth.
- [ ] **Information Architecture spec:** link to CAIA IA feedback, such as an IA spec.
  
##### Platform actions
- [x] [Slack thread with VFS team](https://dsva.slack.com/archives/C04AZ8T7XN1/p1758744797003779)
- [x] Meeting date/time: Async 01:30pm - Thursday, October 9, 2025 (Eastern Time - US & Canada)

##### Platform actions

~~Update this ticket with the Zoom recording~~ This was for engineering and security only, no Governance team members, no meeting took place
  
##### VFS actions
  
- [x] Review the findings tickets and comment on the ticket if there are any questions or concerns
- [ ] Close individual findings tickets when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment on the issue ticket.
- [ ] After launch, [request an accessibility audit from the VA 508 Office](https://depo-platform-documentation.scrollhelp.site/developer-docs/request-support-from-the-va-508-office#RequestsupportfromtheVA508office-AuditRequest). This is required even if no accessibility issues were found during the Staging Review.
  - [ ] Share ServiceNow ticket number here: ______
- [ ] Close ticket once Privacy, Security, Infrastructure Readiness Review has been completed, VA 508 Office audit is requested, and all other post-Staging actions are complete
- [ ] [Complete Collaboration Cycle feedback survey](https://ows.io/qs/gb1dfnkl)

</details>

**Comment by va-albers** [comment]
Based on our discussion we have some takeaways:

- [ ] Steve will discuss the different workload with @little-oddball , with the expectation that we may want to verify that we observe the impact of the service on the production k8s system.  Expectations are that this will be a small impact and the service is already separated from the main vets-api application (it is a Python/FastAPI app).  
- [ ] Steve will start a discussion with the platform security team and @amo-va-octo regarding the potential for PII/PHI included in the model.  This includes a question of how to store the model for daily redeploy.  _Follow-up: the platform security team will set up a meeting to discuss._  
- [ ] @amo-va-octo is going to look into the possibility of using Platform resources for running ML workloads like this (which was currently run on GFE).  This isn't a requirement for this project, but seems like a good thing to consider given this will be an ongoing concern.  
- [ ] Additionally there was an ask for the team to document any manual processes that were used to create the model in question.

None of these should stop continued progress on the project.

**Comment by amo-va-octo** [comment]
Adding a few summarizing the engineering and platform security engineering feedback from this review to help the team ensure it has completed all required action items:

1. Due to concerns of potential exposure to code injection, the team was advised to use a more-secure model serialization format than Python's insecure `pickle` format. `ONNX` was provided as an example of a safer model serialization format, `sklearn-onnx` was found to be compatible with the team's needs in follow-up discussions (https://onnx.ai/sklearn-onnx/)
2. Due to the team's identified concerns for PII to be present in a trained model, the team was asked to store the serialized model file that will be used for inference in a secure S3 bucket instead of adding the model to the associated code repository for the inference/classification server. Access to this bucket should be limited to the service account associated with the inference service.
3. The team was asked to ensure/verify that data cleaning/feature extraction code used in the training set preparation process or pipeline used to train this model be version controlled, and that the data cleaning/feature extraction implementation used in the inference/classification server stay aligned with the implementation and version used in model training.
